src/Interpreters/InterpreterSelectWithUnionQuery.cpp

```diff
@@ -130,7 +130,7 @@ using CustomizeASTSelectWithUnionQueryNormalizeVisitor
 
 InterpreterSelectWithUnionQuery::InterpreterSelectWithUnionQuery(
     const ASTPtr & query_ptr_, const Context & context_, const SelectQueryOptions & options_, const Names & required_result_column_names)
-    : IInterpreterUnionOrSelectQuery(query_ptr_, context_, options_)
+    : IInterpreterUnionOrSelectQuery(query_ptr_->clone(), context_, options_)
```

Comment:
Could you please provide some info - why clone started to be required?

Reply:
Because `query_ptr` also hold by explain query, and here https://github.com/ClickHouse/ClickHouse/blob/04f5b723912e4aef96a71154bb8d5e4ed8c8fcdb/src/Interpreters/InterpreterSelectWithUnionQuery.cpp#L147-L153

changed `query_ptr`, leads to crash.

Reply:
This leads to lots of other `EXPLAIN` releated tests failed(output changed), I will find better way to fix this.